crates/biome_json_analyze/src/lint/nursery/use_required_scripts.rs

```diff
@@ -0,0 +1,145 @@
+use biome_analyze::{Ast, Rule, RuleDiagnostic, context::RuleContext, declare_lint_rule};
+use biome_console::markup;
+use biome_json_syntax::JsonRoot;
+use biome_rowan::{AstNode, AstSeparatedList};
+use biome_rule_options::use_required_scripts::UseRequiredScriptsOptions;
+use rustc_hash::FxHashSet;
+
+declare_lint_rule! {
+    /// Enforce the presence of required scripts in package.json.
+    ///
+    /// This rule ensures that specified scripts are defined in the `scripts` section of a `package.json` file.
+    /// It's particularly useful in monorepo environments where consistency across workspaces is important.
+    ///
+    /// ## Examples
+    ///
+    /// ### Invalid
+    ///
+    /// ```json
+    /// {
+    ///     "scripts": {
+    ///         "test": "vitest"
+    ///     }
+    /// }
+    /// ```
+    ///
+    /// With options `{ "requiredScripts": ["test", "build"] }`, this is invalid because `build` is missing.
+    ///
+    /// ### Valid
+    ///
+    /// ```json
+    /// {
+    ///     "scripts": {
+    ///         "test": "vitest",
+    ///         "build": "tsc"
+    ///     }
+    /// }
+    /// ```
+    ///
+    /// With options `{ "requiredScripts": ["test", "build"] }`, this is valid because all required scripts are present.
+    ///
+    /// ## Options
+    ///
+    /// The rule accepts an array of script names that must be present:
+    ///
+    /// ```json,options
+    /// {
+    ///     "options": {
+    ///         "requiredScripts": ["test", "build", "lint"]
+    ///     }
+    /// }
+    /// ```
+    ///
```

Comment:
Please adhere to the existing patters for documenting options. It's like almost there, but not quite. For example, we have a `use_options` directive that tells the rule linter to use options when linting the docs, and that is not present. Also `requiredScripts`​ should have it's own header under `Options`​. See how other rules are documented.